crates/starknet-devnet-server/src/api/http/models.rs

```diff
@@ -51,7 +52,8 @@ pub struct CreatedBlock {
 #[serde(deny_unknown_fields)]
 #[cfg_attr(test, derive(Debug))]
 pub struct AbortingBlocks {
-    pub(crate) starting_block_hash: BlockHash,
+    pub(crate) starting_block_number: Option<BlockNumber>,
```

Comment:
I think that JSON-RPC parameter should have name `starting_block_block_id` (BlockId type), also here we would end up with starting_block_block of BlockId type.

So user can use hash, number or tag.

In case of aborting pending block, pending block is mined and then aborted. @FabijanC what do you think about this behavior?

Reply:
I had a prepared comment that was about to be posted in the documentation file on the note about block id precedence, but it's basically the same as yours:


IMO, if a user provides both, the most probable cause is that they did it unintentionally and an error should be reported. The best, though a breaking change, would be to have just one property - a `block_id`, similar to what many of Starknet's JSON-RPC methods require:
https://github.com/starkware-libs/starknet-specs/blob/76bdde23c7dae370a3340e40f7ca2ef2520e75b9/api/starknet_api_openrpc.json#L29

Perhaps that would also include block tag (latest/pending) support, not sure how much more complicated it would be for this contribution.

Reply:
about the latest/pending support, I think it's doable, just pending require mine and abort

Reply:
That is how I implemented it first, but deserialization for BlockId type didn't work for numbers or hashes, just worked on tags. Decided to switch from that 1) there is no support to abort on tag and 2) this elegantly retains backwads compatibility without need for `starting_block_hash` to be depreciated in the future.

Reply:
Handled in: 3786ec2751adab30d690599a53dadf871814fadc